portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -8776,6 +8780,19 @@ private String _getVirtualHostname(
 		return virtualHostnames.firstKey();
 	}
 
+	private boolean _isConfiguringPortletEmbedded(
+		LayoutTypePortlet layoutTypePortlet, PortletDisplay portletDisplay,
+		String portletId) {
+
+		if (layoutTypePortlet.isPortletEmbedded(portletId) &&
+			!portletDisplay.isModeView()) {
```

Comment:
Hi @linolaoj, I have the below concerns:

 1. If adding the code **!portletDisplay.isModeView()** in if condition, its portlet title won't be changed. For example, after changing the embedded portlet DDMFormPortlet configuration for **Global**, the expected portlet title should be "Form (Global)" when viewing the layout. I think we may remove the code. 
 
 2. I think you may directly use the layout.isPortletEmbedded(String portletId, long groupId), instead of creating the private method.
 
3. Could you please add one integration test for the case? You may place the integration test in the module com.liferay.portal.osgi.web.portlet.container.test.

Reply:
Hi @yuhai , thank you for looking this, I'll make the changes and add the test.